ecommerce/api.py

```diff
@@ -556,9 +590,53 @@ def import_wire_transfer(wire_transfer, header_row):
         start_date__lte=bootcamp_start_date + timedelta(days=1),
     )
     application = BootcampApplication.objects.get(user=user, bootcamp_run=bootcamp_run)
+    data = {header_row[col]: value for col, value in enumerate(wire_transfer.row)}
+
+    wire_transfer_receipt = WireTransferReceipt.objects.filter(
+        wire_transfer_id=wire_transfer.id
+    ).first()
+    if wire_transfer_receipt:
+        log.info(
+            "Wire transfer %d already imported, checking for the updates...",
+            wire_transfer.id,
+        )
+
+        difference = wire_transfer_difference(wire_transfer_receipt.data, data)
+
+        if difference["order_fields"] and forced:
+            order = wire_transfer_receipt.order
+            line = Line.objects.get(order=order)
+            line.bootcamp_run = bootcamp_run
+            line.description = f"Wire transfer payment for {bootcamp_run}"
+            line.price = wire_transfer.amount
+            line.save()
+
+            previous_application = None
+            if user != order.user or bootcamp_run != order.application.bootcamp_run:
+                previous_application = BootcampApplication.objects.get(
+                    user=order.user, bootcamp_run=order.application.bootcamp_run
+                )
+
+            order.total_price_paid = wire_transfer.amount
+            order.application = application
+            order.amount = wire_transfer.amount
+            order.user = user
+            order.save()
+
+            update_application(application, order)
+            update_application(previous_application, None)
+            log.info("Order data is updated")
+
+            wire_transfer_receipt.data = data
+            wire_transfer_receipt.save()
+            log.info("Receipt data is updated")
+        elif difference["order_fields"] and not forced:
+            raise CommandError("Use --force flag to update the order.")
+        elif difference["receipt_fields"]:
+            wire_transfer_receipt.data = data
+            wire_transfer_receipt.save()
+            log.info("Receipt data is updated")
```

Comment:
Would be good to have some unit tests covering these conditions.

Reply:
I've been going back and forth about this `--force` feature, because it is so dangerous and I don't know yet if I'll need it. If we keep it, it should have tests. But if that's going to take more than say 4 hours of work, i suggest we should just remove the feature and fail on updating a row with order changes. Does that make sense @umarmughal824?